bundles/org.eclipse.swt/Eclipse SWT Browser/win32/org/eclipse/swt/browser/Edge.java

```diff
@@ -489,29 +511,38 @@ public void create(Composite parent, int style) {
 void browserDispose(Event event) {
 	containingEnvironment.instances.remove(this);
 
-	if (webView_2 != null) webView_2.Release();
 	if (environment2 != null) environment2.Release();
-	settings.Release();
-	webView.Release();
-	webView_2 = null;
+	if (settings != null) settings.Release();
+	if(lastWebViewTask.isCancelled()) {
+		// If Someone cancels lastWebViewTask, the webView2 and the webView2_2 would
+		// never be released.
+		SWT.error(SWT.ERROR_DEVICE_DISPOSED);
+	}
+	lastWebViewTask.thenAccept(it -> {
+		it.Release();
+		if (!lastWebView_2Task.isCancelled()) lastWebView_2Task.join().Release();
+		lastWebView_2Task = null;
+	 	lastWebViewTask = null;
+	});
 	environment2 = null;
 	settings = null;
-	webView = null;
-
-	// Bug in WebView2. Closing the controller from an event handler results
-	// in a crash. The fix is to delay the closure with asyncExec.
-	if (inCallback) {
-		ICoreWebView2Controller controller1 = controller;
-		controller.put_IsVisible(false);
-		browser.getDisplay().asyncExec(() -> {
-			controller1.Close();
-			controller1.Release();
-		});
-	} else {
-		controller.Close();
-		controller.Release();
+
+	if(controller != null) {
+		// Bug in WebView2. Closing the controller from an event handler results
+		// in a crash. The fix is to delay the closure with asyncExec.
+		if (inCallback) {
+			ICoreWebView2Controller controller1 = controller;
+			controller.put_IsVisible(false);
+			browser.getDisplay().asyncExec(() -> {
+				controller1.Close();
+				controller1.Release();
+			});
+		} else {
+			controller.Close();
+			controller.Release();
+		}
+		controller = null;
 	}
```

Comment:
This should also be part of the async execution, shouldn't it? It will asynchronously be initialized in `setupBrowser`, so you may come to this line of code while the browser is still being initialized, thus leaving a controller behind that is created after this dipose method has been processed.